### PR TITLE
[enh] Dynamic reports refactored following emdupre guidelines

### PR DESCRIPTION
**Pull request by javiergcas**
Closes # .

Changes proposed in this pull request:

- This code has the structure Elizabeth suggested, which means
- Event links outside function so calls to plotting functions are simplified
- Single function to generate all ranked plots
- Pie plot included
- Variables are all now lower case and match as much as possible Elizabeth's code
- Hovering is now formatted better (only 2 decimal digits, units shown when possible)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -104,4 +104,8 @@ ENV/
 .mypy_cache/
 
 # vscode
-.vscode
\ No newline at end of file
+.vscode
+
+# jupyter notebooks
+.ipynb_checkpoints/
+*.ipynb
```
[the remark this answers is not in the document]
Comment (javiergcas) [reply .gitignore]: Sounds good to me removing the notebooks. In fact, once the code is integrated with the rest of tedana, no notebook should remain in the source code

File: tedana/reporting/data/html/report_head_template.html [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+            <li class="pure-menu-item"><a href="https://twitter.com/dog_rates/status/1193936480944689157?s=20"
+                    target="_blank" class="pure-menu-link">Cute dog</a></li>
+        </ul>
+    </div>
+</nav>
+
+<body>
+    $body
+</body>
+</html>
```
[the remark this answers is not in the document]
Comment (javiergcas) [reply tedana/reporting/data/html/report_head_template.html]: I did not create the templates. This was all @emdupre and @smoia . So unfortunately, I can't comment on this one.

File: .gitignore (hunk 1 above)
[the remark this answers is not in the document]
Comment (emdupre) [reply .gitignore]: I think you're just suggesting that we add a blank line to the end of the file here, @smoia  ? That's not necessary for .gitignore files (but it is PEP8 best-practice for .py files !).

File: tedana/reporting/data/html/report_head_template.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (emdupre) [reply tedana/reporting/data/html/report_head_template.html]: Same comment, it looks like you're just suggesting adding a blank line ? Not best-practice for HTML.

File: tedana/reporting/KappaRho_DynPlot.py [hunk 3]
```diff
@@ -0,0 +1,578 @@
+# ---
```
Comment (emdupre) [inline tedana/reporting/KappaRho_DynPlot.py]: @javiergcas I think this file should be removed and its functionality refactored into the dynamic_figures.py file !

File: tedana/reporting/dynamic_figures.py [hunk 4]
```diff
@@ -0,0 +1,384 @@
...
+    var text = div.text.concat(line);
+    var lines = text.split("\\n")
+        if (lines.length > 35)
+            lines.shift();
+    div.text = lines.join("\\n");
+    """)
+
+
+# %%
+# ----------------------------------------------------------------
```
Comment (emdupre) [inline tedana/reporting/dynamic_figures.py]: This section was me trying to get something to quickly test and show. we should remove everything below L335, here.

**Comment by emdupre** [comment]
> I wouldn't remove the possibility of making figures, just making them not the default (as it was before). The default can be the report, but while that's a great instrument for single subjects and deeper understanding of the results, it's really not helpful in the case of big datasets (multiple subjects/sessions/runs), cause then the intention is to have a good summary of your results to go through with the least time spent on user interface actions (clicks/navigation/...).

By "figures" I'll assume you mean static figures, here?

If so, the brainsprite directories should still provide static figures with visualizations of the components, just not their time series or FFT.

**Comment by smoia** [comment]
Yes, but spatial maps are not enough to interpret the components. In order to evaluate an IC, the minimal information needed is timeseries and/or FFT and spatial maps. I would even say timeseries are more informative than maps sometimes.
The report is an amazing tool to (1) link all the steps of the analysis together, (2) be able to investigate in a bit more deep manner ME related properties of the components and (3) obtain something that is great for presentations.
The figures are still a great tool to (1) go through a rich dataset knowing what the rich dataset is representing and (2) going through a rich dataset in a fairly quick way - much quicker than the report or an external tool such as AFNI or fsleyes.
While I believe that the report could be the default output, I don't think we shouldn't be open to different ways to view the results of MEICA. I see how the possibility of opening the results in external tools is not a tedana problem (although we can facilitate the user in the operation, see #426 ), but the visualisation that is outputted is.
We have the code, and I didn't see (given that I still need to finish the review though) any conflict between having _also_ figures as an output _on request_ and having the report - so I don't see the point not to leave the option of having figures on request.

I might be very passionate about this point because it's fundamental to my preprocessing, but I don't think I'm the only one that would like to go through many components in a way that is between "having a good idea about what a component represents"  and "not loosing too much time in navigating a 3d representation and/or setting up the visualisation".

Yet again, if you prefer the figures output to be a post-tedana output (although I don't see the reason for that), I'm sure that @dowdlelt and I can work on making it that way.

**Comment by emdupre** [comment]
> Yet again, if you prefer the figures output to be a post-tedana output (although I don't see the reason for that), I'm sure that @dowdlelt and I can work on making it that way.

I'd prefer we don't immediately leap to that as a discussion point -- I don't think it's productive to carry the conversation forward.

I also don't think these points strongly relate to the actual code under consideration (no points to line numbers, etc) so I'll ask that we revisit this discussion once the code firms up a bit. 

In general, I'm all for passionate discussions in pull requests but let's keep them productive and focused on the code.

File: tedana/reporting/dynamic_figures.py (hunk 4 above)
Comment (javiergcas) [reply tedana/reporting/dynamic_figures.py]: @emdupre: I think I'm a little confused -- we don't want both the KappaRho_DynPlot.py file and the dynamic_figures.py file. If KappaRho_DynPlot is intended to supersede dynamic_figures, can we remove the current dynamic_figures file and rename KappaRho_DynPlot as "dynamic_figures" ? Correct, KappaRho_DynPlot supersedes dynamic_figures

File: tedana/reporting/KappaRho_DynPlot.py (hunk 3 above)
Comment (javiergcas) [reply tedana/reporting/KappaRho_DynPlot.py]: Correct. Sorry I just commented below in the wrong comment for this question

File: tedana/reporting/data/html/report_head_template.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (smoia) [reply tedana/reporting/data/html/report_head_template.html]: Sorry, my bad - you're right, just suggesting padding with a blank line. This should only be done in python scripts though, right? Are we enforcing it or no problem if we don't?

**Comment by emdupre** [comment]
@62442katieb 's PR for linked brainsprites ! : https://github.com/emdupre/tedana/pull/21/files

**Comment by emdupre** [comment]
This was superseded by #457, so I'm going to close it ! Thanks everyone for your contributions, here !!

**Pull request closed without merging**